ghost/members-api/lib/repositories/member.js

```diff
@@ -726,9 +729,21 @@ module.exports = class MemberRepository {
         const memberIds = memberRows.map(row => row.id);
 
         if (data.action === 'unsubscribe') {
-            return await this._Member.bulkDestroy(memberIds, 'members_newsletters', {column: 'member_id'});
+            const hasNewsletterSelected = (Object.prototype.hasOwnProperty.call(data, 'newsletter') && data.newsletter !== null);
+            if (hasNewsletterSelected) {
+                const newsletters = await this._MemberNewsletter.where('newsletter_id', data.newsletter).fetchAll();
+                const toUnsubscribe = newsletters.toJSON().reduce((acc, newsletter) => {
+                    if (memberIds.includes(newsletter.member_id)) {
+                        acc.push(newsletter.id);
+                    }
+                    return acc;
+                }, []);
+                return await this._Member.bulkDestroy(toUnsubscribe, 'members_newsletters', {column: 'id'});
+            }
+            if (!hasNewsletterSelected) {
+                return await this._Member.bulkDestroy(memberIds, 'members_newsletters', {column: 'member_id'});
```

Comment:
What would be really cool is if we could update the `bulkDestroy` method to make calls like this nicer, imagine if we could instead do:
```js
this._MemberNewsletter.bulkDestroy({member_id: memberIds});


// bulk-operations.js

async bulkDestroy(data, tableName, options) {
  if (!Array.isArray(data)) {
    const [key, ...keys] = Object.keys(data);

    assert(keys.length === 0);
    return this.bulkDestroy(data[key], tableName, {column: key});
  }
  // ...
}
```

Not necessarily in scope for this PR but would be much nicer!